tests/v2/validation/steve/vai/secret_sort_test_cases.go

```diff
@@ -0,0 +1,185 @@
+package vai
+
+import (
+	"fmt"
+	"net/url"
+
+	namegen "github.com/rancher/shepherd/pkg/namegenerator"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type secretSortTestCase struct {
+	name             string
+	createSecrets    func() ([]v1.Secret, []string, []string)
+	sort             func() url.Values
+	expectedOrder    []string
+	supportedWithVai bool
+}
+
+func (s secretSortTestCase) SupportedWithVai() bool {
+	return s.supportedWithVai
+}
+
+var secretSortTestCases = []secretSortTestCase{
+	{
+		name: "Sort by name ascending",
+		createSecrets: func() ([]v1.Secret, []string, []string) {
+			suffix := namegen.RandStringLower(randomStringLength)
+			ns1 := fmt.Sprintf("namespace1-%s", suffix)
+			ns2 := fmt.Sprintf("namespace2-%s", suffix)
+			secrets := []v1.Secret{
+				{ObjectMeta: metav1.ObjectMeta{Name: "secret1", Namespace: ns1}},
+				{ObjectMeta: metav1.ObjectMeta{Name: "secret2", Namespace: ns1}},
+				{ObjectMeta: metav1.ObjectMeta{Name: "secret3", Namespace: ns2}},
+				{ObjectMeta: metav1.ObjectMeta{Name: "secret4", Namespace: ns2}},
+			}
+			return secrets, []string{"secret1", "secret2", "secret3", "secret4"}, []string{ns1, ns2}
+		},
+		sort: func() url.Values {
+			return url.Values{"sort": []string{"metadata.name"}}
+		},
+		expectedOrder:    []string{"secret1", "secret2", "secret3", "secret4"},
```

Comment:
can we use some sort function to sort these names and then send them out as expected instead of manually adding the names for these secrets as secret1, secret2 etc?

Reply:
sure thing, adding:

```go
func getSortedSecretNames(secrets []v1.Secret, sortFunc func() url.Values) []string {
	sortValues := sortFunc()
	sortFields := strings.Split(sortValues.Get("sort"), ",")

	sort.Slice(secrets, func(firstIndex, secondIndex int) bool {
		for _, field := range sortFields {
			ascending := true
			if strings.HasPrefix(field, "-") {
				ascending = false
				field = field[1:]
			}

			var firstValue, secondValue string
			switch field {
			case "metadata.name":
				firstValue, secondValue = secrets[firstIndex].Name, secrets[secondIndex].Name
			case "metadata.namespace":
				firstValue, secondValue = secrets[firstIndex].Namespace, secrets[secondIndex].Namespace
			default:
				continue
			}

			if firstValue != secondValue {
				if ascending {
					return firstValue < secondValue
				}
				return firstValue > secondValue
			}
		}
		return false
	})

	sortedNames := make([]string, len(secrets))
	for index, secret := range secrets {
		sortedNames[index] = secret.Name
	}
	return sortedNames
}
```